Homework3/task_2.py

```diff
@@ -0,0 +1,215 @@
+import os
+import logging
+import csv
+import requests
+import argparse
+import shutil
+from datetime import datetime, timedelta
+from statistics import mean
+from collections import Counter
+
+URL = 'https://randomuser.me/api/?results=5000&format=csv'
+
+
+def setup_logger():
+    logging.basicConfig(
+        filename='my_log_file.log',
+        level=logging.DEBUG,
+        format='%(asctime)s - %(levelname)s - %(message)s"'
+    )
+    return logging.getLogger('my_logger')
+
+
+my_logger = setup_logger()
+
+
+def download_user_accounts_csv(output_csv):
+    my_logger.info('Downloading user accounts data from the API...')
+    response = requests.get(URL)
+    if response.status_code == 200:
+        data = response.text
+        if data:
+            my_logger.info('Received user accounts from the API.')
+            with open(output_csv, 'w', encoding='utf-8') as file:
+                file.write(data)
+            my_logger.info(f'Successfully downloaded user accounts to {output_csv}')
+        else:
+            my_logger.warning('No data received from the API.')
+    else:
+        my_logger.error(f'Failed to fetch data. Status code: {response.status_code}')
+
+
+def read_data_from_csv():
+    with open('user_accounts.csv', 'r', encoding='utf-8') as csvfile:
+        return list(csv.DictReader(csvfile))
+
+
+def write_data_in_csv(file_path, input_data):
+    with open(file_path, 'w', newline='', encoding='utf-8') as csvfile:
+        fieldnames = list(input_data[0].keys())
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        writer.writeheader()
+        writer.writerows(input_data)
+        my_logger.info(f'Data was written in {file_path}.')
+
+
+def add_index_field(input_data):
+    for index, data in enumerate(input_data, start=1):
+        data['global_index'] = index
+    return input_data
+
+
+def add_timezone_offset(datetime_str, offset_str):
+    hours, minutes = map(int, offset_str.split(':'))
+    offset = timedelta(hours=hours, minutes=minutes)
+
+    dt_with_offset = datetime_str + offset
+    return dt_with_offset.strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
I would prefer such way
```suggestion
def add_timezone_offset(datetime_str, offset_str):
    hours, minutes = map(int, offset_str.split(':'))
    dt_with_offset = datetime_str + timedelta(hours=hours, minutes=minutes)
    return dt_with_offset.strftime('%Y-%m-%d %H:%M:%S')
```
or event without dt_with_offset, just return. But you should decide which way you personally like more